nipoppy/sample_run_nipoppy.py

```diff
@@ -131,75 +186,101 @@
                            "/swi/",
                            "/perf"]
          # Run fmriprep tracker to regenerate bagel
-        run_tracker.run(global_configs, dash_schema_file, ["fmriprep"], logger=logger)
+        run_tracker.run(global_configs, dash_schema_file, [wf], session_id=session_id, logger=logger)
 
-        proc_participants, _ = get_new_proc_participants(global_configs, session_id, pipeline="fmriprep", logger=logger)
+        proc_participants, _ = get_new_proc_participants(global_configs, session_id, pipeline=wf, logger=logger)
         n_proc_participants = len(proc_participants)
         
         if n_proc_participants > 0:
-            # remove old bids_db
-            sql_db_file = f"{DATASET_ROOT}/proc/bids_db_fmriprep/bids_db.sqlite" 
-            logger.info(f"Removing old bids_db from {sql_db_file}")
-            if os.path.exists(sql_db_file):
-                os.remove(sql_db_file)
-
-            logger.info(f"Running fmriprep on {n_proc_participants} participants from session: {session} and for modalities: {modalities}")
-            # The default bids_db_path is proc/bids_db_{pipeline}
-            bids_db_path = generate_pybids_index(global_configs, session_id, pipeline="fmriprep", 
-                                                    ignore_patterns=ignore_patterns, logger=logger)
-            logger.info(f"bids_db_path: {bids_db_path}")
-
-            # Don't run more than MAX_BATCH participants in parallel
-            # Also clean-up after MAX_BATCH participants to avoid storage issues
-            if n_proc_participants > MAX_BATCH: 
-                n_batches = int(np.ceil(n_proc_participants/MAX_BATCH))
-                logger.info(f"Running fmriprep in {n_batches} batches of at most {MAX_BATCH} participants each")
-                proc_participant_batches = np.array_split(proc_participants, n_batches)
+            bids_db_path = refresh_bids_db(global_configs, session_id, wf, ignore_patterns, logger)
 
-            else:
-                proc_participant_batches = [proc_participants]
+            # Generate a list of participants to run on HPC 
+            if use_hpc:                
+                hpc_job_list_file = f"{hpc_job_list_dir}/hpc_job_list_{wf}_{session}.txt"
+                logger.info(f"Generating HPC job list for {n_proc_participants} participants: {hpc_job_list_file}")
+                pd.DataFrame(data=proc_participants).to_csv(hpc_job_list_file, header=False, index=False)
```

Comment:
Yes at first glance it looks like a common call across workflows. But `get_new_proc_participants` depends on the updated bagel - which is called on previous line separately for each proc_pipe.

Reply:
I see, but isn't it the same call to run trackers too? So we could just add `run_tracker.run(global_configs, dash_schema_file, [wf], session_id=session_id, logger=logger)` above `get_new_proc_participants` in my previous comment

Reply:
That would have been ideal, but non-`proc_pipe` workflows don't have calls for tracker / bagels ...

Reply:
Hence the top-most `if wf in proc_pipes` condition ;) this would only apply to proc pipes

Reply:
Ahh true - that could work. I guess I have been in two minds since TractoFlow didn't have a tracker until now and maget-brain still doesn't. Also heudiconv does have one now, but I am not sure if this one needs to be called from `run_nipoppy.py`. So let's keep it the way it is for now, and optimize when this orchestration script stabilizes.